pulp_container/app/tasks/sync_stages.py

```diff
@@ -349,6 +385,75 @@ def create_blob(self, blob_data, deferred_download=True):
 
         return blob_dc
 
+    async def create_signatures(self, man_dc, signature_source):
+        """
+        Create signature declarative artifacts.
+
+        Signatures are currently supported only for image manifests and not manifest lists.
+
+        For sigstore, signatures are downloaded from a specified url. The number of signatures is
+        unknown upfront, need to download them in order by incrementing the index until hit 404.
+
+        Args:
+            man_dc: Declarative content instance of a related Manifest
+            signature_source: Source where to get signatures from
+        """
+        signature_dcs = []
+        if signature_source == SIGNATURE_SOURCE.SIGSTORE:
+            man_digest_reformatted = man_dc.content.digest.replace(":", "=")
+            sig_relative_baseurl = f"{self.remote.upstream_name}@{man_digest_reformatted}"
+
+            signature_counter = 1
+            while True:
+                signature_url = urlpath_sanitize(
+                    self.remote.sigstore, sig_relative_baseurl, f"signature-{signature_counter}"
+                )
+                signature_downloader = self.remote.get_noauth_downloader(url=signature_url)
+                try:
+                    signature_download_result = await signature_downloader.run()
+                except aiohttp.client_exceptions.ClientResponseError as exc:
+                    if exc.status != 404:
+                        log.info(
+                            _(
+                                "{} is not accessible, can't sync an image signature. "
+                                "Error: {} {}".format(signature_url, exc.status, exc.message)
+                            )
+                        )
+                    # 404 is fine, it means there are no or no more signatures available
+                    break
+
+                with open(signature_download_result.path, "rb") as f:
+                    signature_raw = f.read()
+
+                signature_counter += 1
+                signature_json = extract_data_from_signature(signature_raw, man_dc.content.digest)
+                if signature_json is None:
+                    continue
+
+                sig_digest = hashlib.sha256(signature_raw).hexdigest()
+                signature = ManifestSignature(
+                    name=f"{man_dc.content.digest}@{sig_digest[:32]}",
+                    digest=sig_digest,
```

Comment:
can you prepend 'sha256'?
```
sig_digest = hashlib.sha256(data).hexdigest()
signature_json["digest"] = f"sha256:{sig_digest}"
```

Reply:
I think you meant to prepend the manifest digest and not the signature digest.
Please check the changes if they look ok now.

Reply:
Never mind, I see this format for manifests.